src/App.js

```diff
@@ -1,43 +1,60 @@
 import * as React from "react";
-import Product from "../src/components/Product";
-import { Box, Grid } from "@mui/material";
+import { Grid } from "@mui/material";
+import ProductList from "./components/ProductList";
+import Navbar from "./components/Navbar";
 import axios from "axios";
+import Cart from "./components/Cart";
+import { Link } from "react-router-dom";
 
 function App() {
+
+  /**********************************************************/
+
   const [products, setProducts] = React.useState([]);
+  const [filteredProducts, setFilteredProducts] = React.useState([]);
+  const [searchInput, setSearchInput] = React.useState("");
+  const [cartItems, setCartItems] = React.useState(0);
+
+  /**********************************************************/
+
 
   React.useEffect(() => {
     axios
       .get(`http://localhost:3000/`)
       .then((res) => {
         setProducts(res.data.product);
+        setFilteredProducts(res.data.product);
       })
       .catch((error) => {
-        console.log(`error`);
+        console.log(`error: ${error}`);
       });
   }, []);
 
+  React.useEffect(() => {
+    setFilteredProducts(
+      products.filter((item) =>
+        item.name.toLowerCase().includes(searchInput.toLowerCase())
+      )
+    );
+  }, [searchInput]);
+
+  /**********************************************************/
+
   return (
-    <Box sx={{ flexGrow: 1 }}>
-      <Grid
-        container
-        spacing={{ xs: 2, sm: 4, md: 3 }}
-        columns={{ xs: 4, sm: 8, md: 12 }}
-      >
-        {products?.map((product) => (
-          <Grid item xs={2} sm={4} md={4} key={product.name}>
-            <Product
-              name={product.name}
-              category={product.category}
-              price={product.price}
-              desc={product.description}
-              stock={product.stock}
-              image_url={product.image_url}
-            />
-          </Grid>
-        ))}
+    <>
+      <Link to="/cart">{Cart}</Link>
+      <Grid>
+        <Grid item>
+          <Navbar setSearchInput={setSearchInput} cartItems={cartItems} />
```

Comment:
To avoid unexpected behavior; a grid `item` should always be wrapped inside a grid `container`
`<Grid container> <Grid item></Grid> </Grid>`